packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -171,19 +197,18 @@ export function useComboBoxState<T extends object>(props: ComboBoxProps<T>): Com
   }, [collection, inputValue, itemsControlled, defaultFilterFn]);
 
   let selectionManager = new SelectionManager(filteredCollection, selectionState);
+  let selectedItem = selectedKey ? collection.getItem(selectedKey) : null;
 
-  // Focus first item if filtered collection no longer contains original focused item
-  useEffect(() => {
-    // Only set a focused key if one existed previously, don't want to focus something by default if customValue = true
-    if (selectionManager.focusedKey && !filteredCollection.getItem(selectionManager.focusedKey)) {
-      selectionManager.setFocusedKey(filteredCollection.getFirstKey());
+  // Prevent open operations from triggering if there is nothing to display
+  let open = (focusStrategy?) => {
+    if (filteredCollection.size > 0) {
+      triggerState.open(focusStrategy);
```

Comment:
This still suffers from the issue of filteredCollection lagging behind a step. E.g.
- User types Aardvark (filtered collection is of size 1 with Aardvark only)
- User closes menu with escape
- User types an extra k
- `triggerState.open` fires because `open` is called in `onChange` in useCombobox, but this happens before the filteredCollection is recomputed for 'Aardvarkk' on the next render so filteredCollection.size is still greater than 0

Would be nice not to tie open with onChange in useCombobox, still trying to figure out how to do it as a sideeffect of inputValue changing but only if the change was caused by the user typing (don't want menu to open if inputValue is changed by selectedKey changing)